### PR TITLE
fix: LogFunc simplify swaps arguments

### PR DESCRIPTION
**Pull request by erratic-pattern**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion/issues/10359.

## Are these changes tested?
yes

## Are there any user-facing changes?
no

### CODE REVIEW

File: datafusion/functions/src/math/log.rs [hunk 1]
```diff
@@ -283,4 +289,52 @@ mod tests {
...
+        let result = LogFunc::new().simplify(vec![lit(2)], &context).unwrap();
+        match result {
+            ExprSimplifyResult::Simplified(_) => {
+                panic!("Expected ExprSimplifyResult::Original")
+            }
+            ExprSimplifyResult::Original(args) => {
+                assert_eq!(args.len(), 1);
+                assert_eq!(args[0], lit(2));
+            }
+        }
```
Comment (alamb) [inline datafusion/functions/src/math/log.rs:324]: Another way to write this perhaps more concisely and with less indenting could be: 


```suggestion
        let ExprSimplifyResult::Original(args) = result  else {
            panic!("Expected ExprSimplifyResult::Original")
        };
        assert_eq!(args.len(), 1);
        assert_eq!(args[0], lit(2));
```

Comment (alamb) [reply datafusion/functions/src/math/log.rs:324]: (same for the other)

**Review by alamb — APPROVED** [review]
nice find and fix @erratic-pattern  🙏

File: datafusion/functions/src/math/log.rs (hunk 1 above)
Comment (erratic-pattern) [reply datafusion/functions/src/math/log.rs:324]: tbh I always forget `let else` exists 😆

File: datafusion/functions/src/math/log.rs [hunk 2]
```diff
@@ -283,4 +289,44 @@ mod tests {
...
+        let context = SimplifyContext::new(&props).with_schema(schema);
+        // One argument with no simplifications
+        let result = LogFunc::new().simplify(vec![lit(2)], &context).unwrap();
+        let ExprSimplifyResult::Original(args) = result else {
+            panic!("Expected ExprSimplifyResult::Original")
+        };
+        assert_eq!(args.len(), 1);
+        assert_eq!(args[0], lit(2));
+        // Two arguments with no simplifications
+        let result = LogFunc::new()
```
Comment (alamb) [inline datafusion/functions/src/math/log.rs:322]: 😍

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks again @erratic-pattern

**Pull request merged**